src/utils.js

```diff
@@ -14,3 +21,65 @@ export function defaults(obj, ...args) {
   });
   return obj;
 }
+
+/**
+ * Replaces all occurrences of the specified text. Returns a new value with the replacements made.
+ * If React is available, this function supports replacing text with React elements and replacing
+ * values within nested React elements and arrays.
+ *
+ * @param {string|object|Array} interpolated - The value to replace occurrences of the specified text in.
+ * @param {string|RegExp} find - The text or regular expression to search for in the interpolated value.
+ * @param {string|object|Array} replace - The value to replace occurrences of the specified text with.
+ * @returns {string|object|Array} A new value with the specified text replaced.
+ */
+export function replaceValue(interpolated, find, replace) {
+  if (!React) {
+    return interpolated.replace(find, replace);
+  }
+
+  switch (typeof interpolated) {
+    case 'string': {
+      const split = interpolated.split(find);
+
+      if (split.length !== 1 && typeof replace === 'object') {
+        // Replace is an object. Return a React fragment with the replacement.
+
+        const props = {key: `${split[0]}`};
+        const childNodes = [split[0], replace, split[1]];
+        return React.createElement(React.Fragment, props, ...childNodes);
```

Comment:
If we didn't need React below, we could return an array here and remove even the soft dependency.